lib/modules/platform/gitlab/code-owners.ts

```diff
@@ -0,0 +1,58 @@
+import ignore from 'ignore';
+import { regEx } from '../../../util/regex';
+import type { FileOwnerRule } from '../types';
+
+export class CodeOwnersParser {
+  private currentSection: { name: string; defaultUsers: string[] };
+  private internalRules: FileOwnerRule[];
+
+  constructor() {
+    this.currentSection = { name: '', defaultUsers: [] };
+    this.internalRules = [];
+  }
+
+  private changeCurrentSection(line: string): void {
+    const [name, ...usernames] = line.split(regEx(/\s+/));
+    this.currentSection = { name, defaultUsers: usernames };
+  }
+
+  private addRule(rule: FileOwnerRule): void {
```

Comment:
please inline this function, no benefit for a one liner which is only called on one place (perf)